test/integration/isolated/udpingress_test.go

```diff
@@ -0,0 +1,156 @@
+//go:build integration_tests
+
+package isolated
+
+import (
+	"context"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	ktfkong "github.com/kong/kubernetes-testing-framework/pkg/clusters/addons/kong"
+	"github.com/kong/kubernetes-testing-framework/pkg/utils/kubernetes/generators"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/annotations"
+	kongv1beta1 "github.com/kong/kubernetes-ingress-controller/v3/pkg/apis/configuration/v1beta1"
+	"github.com/kong/kubernetes-ingress-controller/v3/pkg/clientset"
+	"github.com/kong/kubernetes-ingress-controller/v3/test"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/integration/consts"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/testlabels"
+)
+
+func TestUDPIngressEssentials(t *testing.T) {
+	// Constants shared in many steps of this test that doesn't change.
+	const udpIngressName = "upd-ingress-essentials"
+	const servicePort = ktfkong.DefaultUDPServicePort
+	const serviceName = "udpecho"
+	testUUID := uuid.NewString()
+
+	// Helpers used in this test.
+	requireNoResponse := func(udpGatewayURL string) {
```

Comment:
Provide the `t *testing.T` arg and pass it down at call site. This way the concrete sub tests is mentioned in the logs and that test is set to failed when things go wrong.

Reply:
You're right, it makes logs nicer